python/ray/experimental/state/state_cli.py

```diff
@@ -112,155 +121,59 @@ def list_state_cli_group(ctx):
-def nodes(ctx, format: str, filter: List[Tuple[str, str]]):
-    url = ctx.obj["api_server_url"]
-    format = AvailableFormat(format)
-    print(
-        get_state_api_output_to_print(
-            list_nodes(
-                api_server_url=url,
-                filters=filter,
-                _explain=_should_explain(format),
-            ),
-            format=format,
-        )
-    )
-
-
-@list_state_cli_group.command()
-@list_format_option
-@list_filter_option
-@click.pass_context
-def jobs(ctx, format: str, filter: List[Tuple[str, str]]):
-    url = ctx.obj["api_server_url"]
-    format = AvailableFormat(format)
-    print(
-        get_state_api_output_to_print(
-            list_jobs(
-                api_server_url=url,
-                filters=filter,
-                _explain=_should_explain(format),
-            ),
-            format=format,
-        )
-    )
-
-
-@list_state_cli_group.command()
-@list_format_option
-@list_filter_option
-@click.pass_context
-def workers(ctx, format: str, filter: List[Tuple[str, str]]):
-    url = ctx.obj["api_server_url"]
+    """
+    # All resource names use '_' rather than '-'. But users options have '-'
+    resource = StateResource(resource.replace("-", "_"))
     format = AvailableFormat(format)
-    print(
-        get_state_api_output_to_print(
-            list_workers(
-                api_server_url=url,
-                filters=filter,
-                _explain=_should_explain(format),
-            ),
-            format=format,
-        )
-    )
 
+    # Get the state API server address from ray if not provided by user
+    api_server_address = address if address else _get_state_api_server_address()
 
-@list_state_cli_group.command()
-@list_format_option
-@list_filter_option
-@click.pass_context
-def tasks(ctx, format: str, filter: List[Tuple[str, str]]):
-    url = ctx.obj["api_server_url"]
-    format = AvailableFormat(format)
-    print(
-        get_state_api_output_to_print(
-            list_tasks(
-                api_server_url=url,
-                filters=filter,
-                _explain=_should_explain(format),
-            ),
-            format=format,
-        )
+    # Create the State API server and put it into context
+    logger.debug(f"Create StateApiClient at {api_server_address}...")
+    client = StateApiClient(
+        api_server_address=api_server_address,
     )
 
-
-@list_state_cli_group.command()
-@list_format_option
-@list_filter_option
-@click.pass_context
-def objects(ctx, format: str, filter: List[Tuple[str, str]]):
-    url = ctx.obj["api_server_url"]
-    format = AvailableFormat(format)
-    print(
-        get_state_api_output_to_print(
-            list_objects(
-                api_server_url=url,
-                filters=filter,
-                _explain=_should_explain(format),
-            ),
-            format=format,
-        )
+    options = ListApiOptions(
+        limit=DEFAULT_LIMIT,  # TODO(rickyyx): should we make this configurable?
```

Comment:
Yes. I think there's an open question for supporting limit & sort-by (from the API doc). I think we can only selectively support limit, but we need some discussion!